src/status_im/contexts/wallet/common/sheets/account_options/view.cljs

```diff
@@ -73,9 +74,19 @@
                                               {:type :positive
                                                :text (i18n/label :t/address-copied)}])
                                 (clipboard/set-string address))}
+        {:icon                :i/qr-code
+         :accessibility-label :show-address-qr
+         :label               (i18n/label :t/show-address-qr)
+         :on-press            #(rf/dispatch [:open-modal :wallet-share-address {:status :share}])}
         {:icon                :i/share
          :accessibility-label :share-account
-         :label               (i18n/label :t/share-account)}
+         :label               (i18n/label :t/share-account)
+         :on-press            (fn []
+                                (rf/dispatch [:hide-bottom-sheet])
+                                (js/setTimeout
+                                 #(rf/dispatch [:wallet/share-account
+                                                {:title share-title :content address}])
+                                 600))}
```

Comment:
@mmilad75, it's not clear to me why the dispatch is delayed by 600ms. What happens if we dispatch immediately? Why 600ms in this case?

Reply:
It was there already, so I didn't remove it. I will check on both android and ios and will remove it if it's unnecessary.

Reply:
Without the settimeout, the modal will be closed immediately after opening

Reply:
600ms is the time needed for the first modal to be closed. After it's closed, another modal can be open

Reply:
@mmilad75, I don't see this problem in my tests. I also moved the hide-bottom-sheet call to the event to clean things up. Notice that this orders the events when using `:fx`. Making two subsequent calls to `rf/dispatch` is not the same. I'm attaching the code diff and video. Maybe it doesn't work in iOS, and in that case it's worth adding a little documentation. 

```diff
modified   src/status_im/contexts/wallet/account/tabs/about/view.cljs
@@ -53,12 +53,8 @@
        {:icon                :i/share
         :accessibility-label :share-address
         :label               (i18n/label :t/share-address)
-        :on-press            (fn []
-                               (rf/dispatch [:hide-bottom-sheet])
-                               (js/setTimeout
-                                #(rf/dispatch [:wallet/share-account
-                                               {:title share-title :content address}])
-                                600))}]]]))
+        :on-press            #(rf/dispatch [:wallet/share-account
+                                            {:title share-title :content address}])}]]]))
 
 (defn view
   []
modified   src/status_im/contexts/wallet/common/sheets/account_options/view.cljs
@@ -81,12 +81,8 @@
         {:icon                :i/share
          :accessibility-label :share-account
          :label               (i18n/label :t/share-account)
-         :on-press            (fn []
-                                (rf/dispatch [:hide-bottom-sheet])
-                                (js/setTimeout
-                                 #(rf/dispatch [:wallet/share-account
-                                                {:title share-title :content address}])
-                                 600))}
+         :on-press            #(rf/dispatch [:wallet/share-account
+                                             {:title share-title :content address}])}
         {:add-divider?        (not show-account-selector?)
          :icon                :i/delete
          :accessibility-label :remove-account
modified   src/status_im/contexts/wallet/events.cljs
@@ -345,7 +345,8 @@
                :linkMetadata    {:title title}}]}
             {:title   title
              :subject title
-             :message content})]]}))
+             :message content})]
+         [:dispatch [:hide-bottom-sheet]]]}))
 
 (rf/reg-event-fx :wallet/initialize
  (fn []
```

[Screen_recording_20240111_152753.webm](https://github.com/status-im/status-mobile/assets/46027/2bbe4516-e1f9-4121-8d8c-6f86496b33b1)

Reply:
@ilmotta I think we need to hide the options bottom sheet (and complete the animation) before showing a new bottom sheet

Reply:
@ilmotta Did you try on ios as well? I tried on ios simulator and it was not working properly

Reply:
I can't test on iOS @mmilad75, just on Android, but on Android it looks perfect without the timer workaround, as I shared in the video.

> @ilmotta I think we need to hide the options bottom sheet (and complete the animation) before showing a new bottom sheet or a screen.

Good point! I don't know the answer to that question @OmarBasem. If that's what designers want you're totally right. I personally would prefer to immediately show the next bottom sheet and ignore the closing animation for the current one because we're closing the previous one anyway, so why make the user wait more.

Or, what I actually prefer was if the previous bottom sheet wasn't closed at all, so that I can go back to where I was. I would prefer if the bottom sheets could be nested from a UX perspective without closing previous ones. While I was testing this feature to share, I disliked the fact that if I gave up on sharing for any reason I would have to press on the `...` icon again.

For screens it's a different story, also for performance reasons.

But good point @OmarBasem, our designers are the final word for this kind of thing.